core/model/modx/processors/element/tv/renders/mgr/input/tag.class.php

```diff
@@ -13,42 +13,43 @@
  * @package modx
  * @subpackage processors.element.tv.renders.mgr.input
  */
-class modTemplateVarInputRenderTag extends modTemplateVarInputRender {
-    public function process($value,array $params = array()) {
-        $value = is_array($value) ? $value : explode(',',$value);
+class modTemplateVarInputRenderTag extends modTemplateVarInputRender
+{
+    /**
+     * @param string|array $value
+     * @param array $params
+     * @return mixed|void
+     */
+    public function process($value, array $params = array())
+    {
+        $value = is_array($value) ? $value : explode(',', $value);
 
-        $default = explode('||',$this->tv->get('default_text'));
+        $options = array();
 
-        $options = $this->getInputOptions();
-        $opts = array();
-        $defaults = array();
-        $i = 0;
-
-        while (list($item, $itemValue) = each ($options)) {
-            $checked = false;
-            $itemValue = is_array($itemValue) ? $itemValue : explode('==',$itemValue);
-            $item = $itemValue[0];
-            $itemValue = isset($itemValue[1]) ? $itemValue[1] : $item;
-            if (in_array($itemValue,$value)) {
-                $checked = true;
-            }
-            if (in_array($itemValue,$default)) {
-                $defaults[] = 'tv'.$this->tv->get('id').'-'.$i;
+        foreach ($this->getInputOptions() as $option) {
+            if (!$option) { continue; }
+            $option = is_array($option) ? $option : explode('==', $option);
+            if (count($option) === 1) {
+                $option[] = $option[0];
             }
-
-            $opts[] = array(
-                'value' => htmlspecialchars($itemValue,ENT_COMPAT,'UTF-8'),
-                'text' => htmlspecialchars($item,ENT_COMPAT,'UTF-8'),
-                'checked' => $checked,
+            list($label, $meaning) = $option;
```

Comment:
Why not $value and $text. The variables are only used for placeholder values afterwards and could use their name.

Reply:
`$value` already comes as the function parameter, I can not redefine it in the loop. About text yes, makes sense but label and meaning look more like a pair that text and meaning.

Reply:
Oh, I did not see that.

Meaning is somehow the wrong word.  What about `$tagText` and `$tagValue`?

Reply:
`$inputOptionText`and `$inputOptionValue`?